### PR TITLE
*: sync with new server/client

### PR DESCRIPTION
**Pull request by alissa-tung**

### CODE REVIEW

**Comment by alissa-tung** [comment]
r+ @s12f @YangKian @Commelina

File: app/src/test/java/io/hstream/testing/BasicTest.java [hunk 1]
```diff
@@ -367,11 +373,11 @@ void testWriteRawBatch() throws Exception {
   void testBatchSizeZero() throws Exception {
     final String streamName = randStream(hStreamClient);
     Producer producer =
-        hStreamClient.newProducer().stream(streamName).enableBatch().recordCountLimit(0).build();
+        hStreamClient.newBufferedProducer().stream(streamName).recordCountLimit(0).build();
```
Comment (s12f) [inline app/src/test/java/io/hstream/testing/BasicTest.java]: recordCountLimit is not allowed to be less than 1, so we should change this test case to assert an expected exception(HStreamDBClientException).

Comment (alissa-tung) [reply app/src/test/java/io/hstream/testing/BasicTest.java]: thx [ad5a784](https://github.com/hstreamdb/integration-tests/pull/29/commits/ad5a784490127844d2699b072ee74a61c08039a3)

File: app/src/test/java/io/hstream/testing/TestUtils.java [hunk 2]
```diff
@@ -49,43 +49,41 @@ public static String randText() {
     return UUID.randomUUID().toString().getBytes(StandardCharsets.UTF_8);
   }
 
+  public static Record randRawRec() {
+    return Record.newBuilder().rawRecord(randBytes()).build();
```
Comment (YangKian) [inline app/src/test/java/io/hstream/testing/TestUtils.java]: return buildRecord(randBytes());

File: app/src/test/java/io/hstream/testing/TestUtils.java [hunk 3]
```diff
@@ -199,9 +197,12 @@ public static boolean isAscending(List<RecordId> input) {
   }
 
   public static void assertRecordIdsAscending(List<ReceivedRawRecord> input) {
-    Assertions.assertTrue(
-        isAscending(
-            input.stream().map(ReceivedRawRecord::getRecordId).collect(Collectors.toList())));
+    if (!input.isEmpty()) {
```
Comment (YangKian) [inline app/src/test/java/io/hstream/testing/TestUtils.java]: no need to check here?  call map on an empty list should return an empty list and will be rejected in `isAscending()`  method

File: app/src/test/java/io/hstream/testing/BasicTest.java [hunk 4]
```diff
@@ -97,13 +101,13 @@ public void teardown() throws Exception {
...
 
-  void produce(Producer p, int tid) throws InterruptedException {
+  void produce(Producer p, int tid) throws Exception {
     Random rand = new Random();
     byte[] rRec = new byte[128];
     for (int i = 0; i < 1000; i++) {
       logger.info("Thread " + tid + " write");
       rand.nextBytes(rRec);
-      p.write(rRec).join();
+      p.write(Record.newBuilder().rawRecord(rRec).build()).join();
```
Comment (YangKian) [inline app/src/test/java/io/hstream/testing/BasicTest.java]: Use the helper function `buildRecord()` in `TestUtil` to avoid writing this directly.

**Review by YangKian — CHANGES_REQUESTED** [review]

File: app/src/test/java/io/hstream/testing/TestUtils.java (hunk 3 above)
Comment (YangKian) [reply app/src/test/java/io/hstream/testing/TestUtils.java]: if you want `assertRecordIdsAscending(emptyList)` success, then modify the implemention of `isAscending()` method.

**Review by YangKian — APPROVED** [review]

**Pull request merged**